### PR TITLE
850 create author details page - testCafe

### PR DESCRIPTION
**Pull request by norisalsaadie**
closes https://github.com/libero/reviewer/issues/850

### CODE REVIEW

File: tests/page-objects/AuthorDetailsPage.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    public async populateForm(): Promise<void> {
+        await this.setFirstName('first');
+        await this.lastNameInput('last');
+        await this.setEmail('email@elifesciences.org');
+        await this.setInstitution('institution');
+    }
+
+    public async setEmail(input: string): Promise<void> {
```
Comment (will-byrne) [inline tests/page-objects/AuthorDetailsPage.ts:26]: Can the setters all have a default, similar to the xpub page objects?

File: tests/page-objects/AuthorDetailsPage.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+export class AuthorDetailsPage {
+    private readonly nextButton = Selector('.submission-wizard-next-button');
+
+    private readonly firstNameInput = Selector('.orcid-details__firstName');
+    private readonly lastNameInput = Selector('.orcid-details__lastName');
+    private readonly emailInput = Selector('.orcid-details__email');
+    private readonly institutionInput = Selector('.orcid-details__institution');
+
+    public async assertOnPage(): Promise<void> {
+        await t.expect(this.firstNameInput.visible).ok();
```
Comment (Cooryd) [inline tests/page-objects/AuthorDetailsPage.ts:12]: We should probably assert on a `.author-step` class or something to the form to make the code easier to refactor in future. For Files we've just added a `.files-step` to the outer div

**Comment by will-byrne** [comment]
There is no code for clicking on the orcid autofill.

File: tests/page-objects/AuthorDetailsPage.ts (hunk 1 above)
Comment (norisalsaadie) [reply tests/page-objects/AuthorDetailsPage.ts:26]: Sure

File: tests/page-objects/AuthorDetailsPage.ts (hunk 2 above)
Comment (norisalsaadie) [reply tests/page-objects/AuthorDetailsPage.ts:12]: Sensible. Was going to do that as part of another PR but equally valid. I'll get it done

**Pull request merged**